bindings/rust/integration/tests/internet_http_client.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Purpose: ensure that s2n-tls is compatible with other http/TLS implementations
+///
+/// This test uses s2n-tls-hyper to make http requests over a TLS connection to
+/// a number of well known http sites.
+#[cfg(feature = "network-tests")]
+mod http_get {
+    use bytes::Bytes;
+    use http::{StatusCode, Uri};
+    use http_body_util::{BodyExt, Empty};
+    use hyper_util::{client::legacy::Client, rt::TokioExecutor};
+    use s2n_tls::{config::Config, security};
+    use s2n_tls_hyper::connector::HttpsConnector;
+    use std::{error::Error, str::FromStr};
+    use tracing_subscriber::filter::LevelFilter;
+
+    #[derive(Debug)]
+    struct TestCase {
+        pub query_target: &'static str,
+        pub expected_status_code: u16,
+    }
+
+    impl TestCase {
+        const fn new(domain: &'static str, expected_status_code: u16) -> Self {
+            TestCase {
+                query_target: domain,
+                expected_status_code,
+            }
+        }
+    }
+
+    const TEST_CASES: &[TestCase] = &[
+        // Akamai hangs indefinitely. This is also observed with curl and chrome
+        // https://github.com/aws/s2n-tls/issues/4883
+        // TestCase::new("https://www.akamai.com/", 200),
+
+        // this is a link to the s2n-tls unit test coverage report, hosted on cloudfront
+        TestCase::new("https://dx1inn44oyl7n.cloudfront.net/main/index.html", 200),
+        // this is a link to a non-existent S3 item
+        TestCase::new("https://notmybucket.s3.amazonaws.com/folder/afile.jpg", 403),
```

Comment:
Would not needing to do a valid HTTP get request make hitting cloudfront and s3 easier? I'm thinking about my comment on akamai.

Reply:
I think it would make it easier, but I'm not sure that's the right tradeoff. HTTP requests to S3/CloudFront make up a pretty sizeable chunk of the internet, so I think this is one of the more important use cases for s2n-tls-hyper.